File: src/server/dto/menuPage/menuPage.go
package menuPage

type MenuPage struct {
	MenuItems []MenuItem
}

type MenuItem struct {
	Name string
	Icon string
	Link string
}

func NewMenuPage() *MenuPage {
	model := MenuPage{}
	model.addItem("Files", "folder", "/files/root")

	return &model
}

func (m *MenuPage) addItem(name string, icon string, link string) {
	m.MenuItems = append(m.MenuItems, MenuItem{
		Name: name,
		Icon: icon,
		Link: link,
	})
}

File: front/src/pages/mainMenu/MenuPage.js
import MainMenuButton from "./components/MainMenuButton";
import { useRequest } from "../../hooks/useRequest";

function MenuPage() {
  const response = useRequest('/')

  return (
    <div className="mp-body">
      {response.data != null ?
        response.data.MenuItems.map((item) => <MainMenuButton key={item.Icon} name={item.Name} icon={item.Icon} link={item.Link}/>) :
        ""  
      }
    </div>
    
  );
}

export default MenuPage;
